packages/server/server/api/trecipes/trecipe.service.ts

```diff
@@ -124,24 +127,33 @@ class TrecipeService {
             )
             .then((trecipeToCopy: Trecipe) => {
                 if (trecipeToCopy) {
-                    const copy: Trecipe = {
-                        uuid: uuid(),
-                        name: trecipeToCopy.name,
-                        description: trecipeToCopy.description,
-                        isPrivate: trecipeToCopy.isPrivate,
-                        owner: trecipeToCopy.owner,
-                        collaborators: trecipeToCopy.collaborators,
-                        image: trecipeToCopy.image,
-                        destinations: trecipeToCopy.destinations.map((dest) => {
-                            return {
-                                destUUID: dest.destUUID,
-                                completed: dest.completed,
-                            };
-                        }),
-                        createdAt: '',
-                        updatedAt: '',
-                    };
-                    return new trecipeModel(copy).save();
+                    const isOwner = trecipeToCopy.owner === user.username;
+                    if (isOwner || !trecipeToCopy.isPrivate) {
+                        const copy: Trecipe = {
+                            uuid: uuid(),
+                            name: trecipeToCopy.name,
+                            description: trecipeToCopy.description,
+                            isPrivate: trecipeToCopy.isPrivate,
+                            owner: user.username,
+                            collaborators: [],
+                            image: trecipeToCopy.image,
+                            destinations: trecipeToCopy.destinations.map((dest) => {
+                                return {
+                                    destUUID: dest.destUUID,
+                                    completed: isOwner ? dest.completed : false,
+                                };
+                            }),
+                            createdAt: '',
+                            updatedAt: '',
+                        };
+                        return new trecipeModel(copy).save();
+                    } else {
+                        Promise.reject(
```

Comment:
I think instead of `InternalServerError` we should be returning Not Authorized here? Since that failure is not our fault LOL

Reply:
copy paste hehe I will update it to Not Authorized error